torch/utils/hooks.py

```diff
@@ -24,28 +25,35 @@ def __init__(self, hooks_dict: Any, *, extra_dict: Any = None) -> None:
         self.id = RemovableHandle.next_id
         RemovableHandle.next_id += 1
 
-        self.extra_dict_ref = (
-            weakref.ref(extra_dict)
-            if extra_dict is not None
-            else None
-        )
+        self.extra_dict_ref: Optional[Union[List[Any], Any]] = None
```

Comment:
I think normalizing self.extra_dict_ref to always be a tuple (without changing the init arg type) will make this code simpler I think.